### ISSUE TITLE
vmalert: api: empty fields for recording rules

### ISSUE THREAD

**Issue by antoinedeschenes** [VictoriaMetrics/VictoriaMetrics#7297]
### Describe the bug

It looks like "recording" rules returned by `/api/v1/rules` in vmalert have a blank `state`, `group_name` and `file` fields at the rule level.

```
      {
        "name": "kube-apiserver.rules",
        "rules": [
          {
            "state": "",
            "name": "cluster_quantile:apiserver_request_duration_seconds:histogram_quantile",
            "query": "histogram_quantile(0.5, sum(rate(apiserver_request_duration_seconds_bucket{service=\"kubernetes\",subresource!=\"log\",verb!~\"LIST|WATCH|WATCHLIST|DELETECOLLECTION|PROXY|CONNECT\"}[5m])) without(instance, pod))\n",
            "duration": 0,
            "keep_firing_for": 0,
            "labels": {
              "quantile": "0.5"
            },
            "lastError": "",
            "evaluationTime": 0.027526081,
            "lastEvaluation": "2024-10-18T12:01:16.321855834Z",
            "health": "ok",
            "type": "recording",
            "datasourceType": "prometheus",
            "lastSamples": 31,
            "lastSeriesFetched": 3768,
            "id": "4534695228641503937",
            "group_id": "4652008536941098091",
            "group_name": "",
            "file": "",
            "debug": false,
            "max_updates_entries": 20
          }
        ],
        "interval": 300,
        "lastEvaluation": "2024-10-18T12:01:16.081883567Z",
        "type": "prometheus",
        "id": "4652008536941098091",
        "file": "/etc/vmalert/config/vm-vmstack-rulefiles-0/victoria-metrics-kubernetes-rules.yaml",
        "concurrency": 1
      }
```

### To Reproduce

1. Start vmalert with recording rules
2. Reach the API at http://localhost:8080/api/v1/rules

### Version

vmalert-20241002-113449-tags-v1.104.0-0-g0d4f4b8f7d

### Logs

None

### Used command-line flags

using the k8s-stack chart

```
    - -datasource.url=http://vmsingle-vmstack.victoria-metrics.svc:8429/
    - -evaluationInterval=5m
    - -http.pathPrefix=/
    - -httpListenAddr=:8080
    - -loggerFormat=json
    - -loggerLevel=WARN
    - -notifier.url=http://vmalertmanager-vmstack.victoria-metrics.svc:9093/
    - -remoteRead.url=http://vmsingle-vmstack.victoria-metrics.svc:8429/
    - -remoteWrite.disablePathAppend=true
    - -remoteWrite.url=http://vmagent-vmstack.victoria-metrics.svc.cluster.local:8429/api/v1/write
    - -rule.resendDelay=4m30s
    - -rule="/etc/vmalert/config/vm-vmstack-rulefiles-0/*.yaml"
```

### Additional information

https://github.com/VictoriaMetrics/VictoriaMetrics/pull/7298

_No response_

**Comment by antoinedeschenes** [comment]
fixing in #7298

**Comment by zekker6** [comment]
The bug fix was released in [v1.97.11](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.97.11), [v1.102.6](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.102.6) and [v1.106.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.106.0). Closing as completed.
